stl/inc/xstring

```diff
@@ -2642,6 +2642,44 @@ private:
     }
 
 public:
+#if _HAS_CXX20
+    void _Move_assign_from_buffer(_Elem* const _Right, const size_type _Size, const size_type _Res) {
+        // Move assign from a buffer, used by basic_stringbuf
+        _Tidy_deallocate();
+        _Right[_Size]  = _Elem();
+        auto& _My_data = _Mypair._Myval2;
+        _Construct_in_place(_My_data._Bx._Ptr, pointer_traits<pointer>::pointer_to(*_Right));
+        _My_data._Mysize = _Size;
+        _My_data._Myres  = _Res;
+    }
+
+    struct _Released_buffer {

```

Comment:
This buffer not having the allocator that goes with it is a little scary; perhaps we should copy the allocator in there too to reduce the likelihood of bugs where the allocators mismatch?

That should get tests.

Reply:
The allocator may not propagate on container move assignment, so copy may not be a right way to do so.

Reply:
Hmmm I'm not sure I understand; there is nothing in P0408 that suggests we engage POCMA. (In fact everything is described in terms of move construction, not move assignment, so I believe we *must not* engage POCMA.)

Reply:
Hmmm maybe not:

```
Effects: Equivalent to:
buf = std::move(s);
```

Although in the non-POCMA non-equal-allocators case, getting a `_Released_buffer` would be incorrect, so I don't see how it matters here.

Reply:
Alright, I'll add an allocator field.

Reply:
Could you check that I do the right thing now? I'm not sure about the copying of the allocator.